cpython-unix/build-cpython.sh

```diff
@@ -426,6 +453,11 @@ if [ "${PYBUILD_PLATFORM}" = "macos" ]; then
     export MACOSX_DEPLOYMENT_TARGET="${APPLE_MIN_DEPLOYMENT_TARGET}"
 fi
 
+# ptsrname_r is only available in SDK 13.4+, but we target a lower version for compatibility.
+if [ "${PYBUILD_PLATFORM}" = "macos" ]; then
+    CONFIGURE_FLAGS="${CONFIGURE_FLAGS} ac_cv_func_ptsname_r=no"
+fi
```

Comment:
See https://github.com/zanieb/python-build-standalone/pull/6#issuecomment-2329715746 for more context.

Reply:
Python 3.13 only supports macOS 10.13+. The official build are built for 10.13+. So I'd think the macOS target version should be Python version dependent.

Reply:
I think we're intentionally setting a lower target deployment for compatibility with older operating systems. We do this in various other places already and CPython 3.13 has code to behave correctly if `ptsname_r` is not available.

We could consider bumping the minimum target version, but I think that should be separate from initial suport.

Reply:
https://pythoninsider.blogspot.com/2024/06/python-3130-beta-2-released.html?m=1

> The minimum supported macOS version was changed from 10.9 to 10.13 (High Sierra). Older macOS versions will not be supported going forward.

Reply:
Well, that's good to know. Regardless, I'm not sure that means that we need to drop support yet. I actually tried to bump the minimum version first and it didn't resolve the commented issue because `ptsname_r` is only available in 10.13.4 not 10.13.0. Changing to a minimum supported patch version is a larger change and one I'm hesitant about it. Separately, there's not support for conditional macOS target versions yet and I'd need to implement that. I'd rather get this out the door first.

Reply:
Also the section you linked references the changes to "the macOS installer we provide" which is bit different than CPython itself no?

Reply:
Might be a question for @ned-deily. I believe officially only 10.13+ is tested and supported now, though it might happen to work for older versions for a while.

Reply:
Since we package the distributions here it should be fairly obvious when it stops working and we can address the problems then, either with patches or by upgrading to a newer version. I think the spirit here is to maintain compatibility with as many systems as feasible, so I'll bias towards that to start.

Reply:
cc @hugovk (if you're interested)

Reply:
The comment about changing the minimum support macOS version from 10.9 to 10.13 does refer to the python.org macOS installer and not necessarily to CPython itself. (At the moment, we don't have an officially published support policy for which macOS releases are supported by CPython although we plan to make one.) But there is no intent to deliberately break support on older releases. The above patch, though, does point out a deficiency in CPython today as @zanieb notes, as there currently is no run-time test for the presence of `ptsname_r`, whose use was introduced fairly recently, which would be needed to maintain full weak-linking support on older systems. It would be good to open a cpython issue about that.

Reply:
Thanks for weighing in @ned-deily!

Reply:
Here's an upstream issue https://github.com/python/cpython/issues/123797

Reply:
I would be fine dropping 10.9 support on 3.13. I think targeting what the official installers support is _good enough_.

If some random user on 10.9-10.2 complains, they can always build things from source. Presumably they've been doing that for years because the ~entire world has dropped support for their OS version years ago.